File: Java/lambda/1/main.java
/**
 * Created by fay on 2017/5/8.
 */

public class main {
    public static void main(String args[]) {
        main m = new main();
        m.say((content) -> System.out.println(content));
    }

    void say(Reply reply) {
        reply.r("Hello World!");
    }
}

interface Reply {
    void r(String content);
}

File: Lua/lambda/1/main.lua
--
-- Created by IntelliJ IDEA.
-- User: fay
-- Date: 2017/5/31
-- Time: 下午3:19
-- To change this template use File | Settings | File Templates.
--

function main()
    say(reply)
end

reply = function(content)
    print(content)
    return "Hello Lua!"
end

function say(reply)
    print(reply("Hello World!"))
end

main()
